stdlib/pathname.rb

```diff
@@ -37,7 +44,7 @@ def cleanpath
   end
 
   def to_path
-    @path
+    @path.to_s
```

Comment:
good idea, I'm just a little bit concerned with `<=>` method but I think it's better to do it in the initialize for consistency.